core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -156,9 +161,11 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
                             diskThresholdSettings.getFreeBytesThresholdHigh(), freeBytes, node.nodeId());
                 }
                 return allocation.decision(Decision.NO, NAME,
-                        "the node is above the high watermark even though this shard has never been allocated " +
-                                "and has less than required [%s] free on node, free: [%s]",
-                        diskThresholdSettings.getFreeBytesThresholdHigh(), new ByteSizeValue(freeBytes));
+                    "the node is above the high watermark [%s=%s] and has less than the required [%s] free space, actual free: [%s] " +
+                        "preventing allocation even though the primary has never been allocated",
```

Comment:
my main reason for keeping it was because it alerts us as to why the high watermark was used instead of the low watermark (b/c the shard was never allocated previously).  However, you are right that it is more confusing to the user, and for expert use cases, we can know that the high watermark was used for a never before allocated shard if the shard is in the unassigned state (as opposed to the other situation when high watermark is used which is when we move a shard from one node to another, in which case it is already assigned).  So I'm good with removing it.